src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests2.cs

```diff
@@ -2596,5 +2596,139 @@ public void M(string s)
                 Diagnostic(ErrorCode.WRN_IsPatternAlways, "s is string or null").WithArguments("string").WithLocation(7, 13)
                 );
         }
+
+        [Fact]
+        public void SemanticModelForSwitchExpression()
```

Comment:
>SemanticModelForSwitchExpression [](start = 20, length = 32)

Please add test for similar scenarios in presence of explicit cast applied to the switch expression